Agents/HeatEmissionAgent/HeatEmissionAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatEmissionAgent.java

```diff
@@ -1,176 +1,73 @@
-				heatemission.append("\"AH_3:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_4:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_5:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_6:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_7:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_8:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_9:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_10:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_11:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_12:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_13:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_14:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_15:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_16:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_17:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_18:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_19:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_20:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_21:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_22:MW\":").append(heat_data).append(",\n");
-				heatemission.append("\"AH_23:MW\":").append(heat_data).append("\n").append("},\n");
-				heatemission.append("\"geometry\": {\n" + "\"type\": \"Point\",\n" + "\"coordinates\": [\n")
-						.append(x_coordi + ",\n").append(y_coordi + "\n]\n}\n},\n");
-				Combined.append(heatemission);
-
+			// Properties of database containing buildings data.
+			String dbUrl = null;
+			String dbUser = null;
+			String dbPassword = null;
+
+			if (requestParams.has("dbUrl")) {
+				dbUrl = requestParams.getString("dbUrl");
+				dbUser = requestParams.getString("dbUser");
+				dbPassword = requestParams.getString("dbPassword");
+			} else {
+				String dbName = requestParams.getString("dbName");
+				EndpointConfig endpointConfig = new EndpointConfig();
+				dbUrl = endpointConfig.getDbUrl(dbName);
+				dbUser = endpointConfig.getDbUser();
+				dbPassword = endpointConfig.getDbPassword();
 			}
 
-			// Output heat emission data in GeoJSON format for DUCT
-
-			String outputFP = Paths.get(System.getProperty("user.dir"), "output", "base.geojson").toString();
-			String output = Combined.substring(0, Combined.length() - 2) + "\n]\n}";
-			File file1 = new File(outputFP);
-			FileWriter fw;
-			try {
-				fw = new FileWriter(file1);
-				PrintWriter pw = new PrintWriter(fw);
-				pw.println(output);
-				pw.close();
-			} catch (IOException e) {
-				e.printStackTrace();
-			}
+			rdbStoreClient = new RemoteRDBStoreClient(dbUrl, dbUser, dbPassword);
+
```

Comment:
What about the remaining industries? i.e. pharma, food, semicon and chemicals? do they all come under factory?

Reply:
The pharmaceutical, food, semiconductors and chemicals plants data are stored in the factories table. The data for the precision engineering, printing and data centres are stored in separate tables. The data for the various industrial facilities is grouped into PostgreSQL tables based on the parameters used to estimate heat emissions. 

The factories table contains data for facilities in multiple industries because the parameters used to estimate heat emissions are the specific energy conusmption and production volume in all cases.